### PR TITLE
Accumulo 4234

### PR DESCRIPTION
**Pull request by ArbaazKhan1**
Added a counter in TableMgmtStats to help keep track of tablets that have errors while running a full scan. 
Fixes [issue#4234](https://github.com/apache/accumulo/issues/4234)

### CODE REVIEW

**Comment by EdColeman** [comment]
Is this targeting an incorrect branch?  Should it be elasticity instead of main?

**Comment by ArbaazKhan1** [comment]
No, It should be in elasticity.

**Comment by ArbaazKhan1** [comment]
I haven't, how would I go about sharing it with the metrics.

**Comment by EdColeman** [comment]
One way would be like: https://docs.micrometer.io/micrometer/reference/concepts/gauges.html#_manually_incrementing_or_decrementing_a_gauge

but not sure that is feasible here.  It may also be possible to "wrap" an AtomicIntger / Long with the metrics registration.  That is, create an Atomic and then pass that to the metrics during registration (rather than creating and using the reference as shown in that reference)

File: server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java [hunk 1]
```diff
@@ -398,6 +399,7 @@ private TableMgmtStats manageTablets(Iterator<TabletManagement> iter,
             "Error on TabletServer trying to get Tablet management information for extent: {}. Error message: {}",
             tm.getExtent(), mtiError);
         this.metrics.incrementTabletGroupWatcherError(this.store.getLevel());
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java:401]: This line increments the metric that indicates that an error occurred in the TabletGroupWatcher.

File: server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java [hunk 2]
```diff
@@ -680,7 +682,7 @@ public void run() {
 
         iter = store.iterator(tableMgmtParams);
         var tabletMgmtStats = manageTablets(iter, tableMgmtParams, currentTServers, true);
-        lookForTabletsNeedingVolReplacement = tabletMgmtStats.totalVolumeReplacements != 0;
+        lookForTabletsNeedingVolReplacement = tabletMgmtStats.totalVolumeReplacements != 0 || tabletMgmtStats.tabletsWithErrors != 0;
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java:685]:
```suggestion
        // Continue to look for tablets needing volume replacement if there was an error processing
        // tablets in the call to manageTablets() or if we are still performing volume replacement. We
        // only want to stop looking for tablets that need volume replacement when we have successfully
        // processed all tablet metadata and no more volume replacements are being performed.
        lookForTabletsNeedingVolReplacement = tabletMgmtStats.totalVolumeReplacements != 0 || tabletMgmtStats.tabletsWithErrors != 0;
```

**Review by dlmarion — APPROVED** [review]
Looks good. I added a suggestion regarding a comment that we could include to note why we are setting the variable

**Review by keith-turner — APPROVED** [review]

**Pull request merged**